lib/htmlbars-plugin/index.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var utils = require('./utils');
+
+function ClassTransformPlugin() {
+  this.syntax = null;
+  this.builders = null;
+}
+
+ClassTransformPlugin.prototype.constructor = ClassTransformPlugin;
+
+ClassTransformPlugin.prototype.transform = function(ast) {
+  if (!this.builders) {
+    this.builders = this.syntax.builders;
+  }
+
+  var plugin = this;
+  var walker = new this.syntax.Walker();
+
+  walker.visit(ast, function(node) {
+    switch(node.type) {
+      case 'ElementNode':
+        plugin.transformElementNode(node);
+        break;
+      case 'MustacheStatement':
+      case 'BlockStatement':
+        plugin.transformStatement(node);
+        break;
+    }
+  });
+
+  return ast;
+};
+
+ClassTransformPlugin.prototype.transformStatement = function(node) {
+  var localClassPair = utils.getPair(node, 'local-class');
+  if (!localClassPair) { return; }
+
+  utils.removePair(node, localClassPair);
+
+  var classPair = utils.getPair(node, 'class');
+  var params = [];
+  var concatSexpr;
+
+  if (classPair) {
+    params.push(classPair.value);
+    utils.removePair(node, classPair);
+  }
+
+  console.log(localClassPair);
+  utils.pushAll(params, this.localToPath(localClassPair.value));
+  this.divide(params);
+  concatSexpr = this.builders.sexpr('concat', params);
```

Comment:
It looks like the corresponding builders don't support it, but `Pair`s and `Subexpression` both [parse with `loc` keys in the usual places](https://tonicdev.com/5758ce37bd5dac1400c233ae/5773d6a644e4961300401456), so you could tag it on after the fact.